Mathlib/Topology/Compactness/Lindelof.lean

```diff
@@ -693,45 +692,196 @@ instance {X : ι → Type*} [Countable ι] [∀ i, TopologicalSpace (X i)] [∀
     rw [Sigma.univ]
     exact isLindelof_iUnion fun i => isLindelof_range continuous_sigmaMk
 
-instance Quot.LindelofSpace {r : X → X → Prop} [LindelofSpace X] : LindelofSpace (Quot r) where
+instance Quot.instLindelofSpace {r : X → X → Prop} [LindelofSpace X] : LindelofSpace (Quot r) where
   isLindelof_univ := by
     rw [← range_quot_mk]
     exact isLindelof_range continuous_quot_mk
 
-instance Quotient.LindelofSpace {s : Setoid X} [LindelofSpace X] : LindelofSpace (Quotient s) :=
-  Quot.LindelofSpace
+instance Quotient.instLindelofSpace {s : Setoid X} [LindelofSpace X] : LindelofSpace (Quotient s) :=
+  Quot.instLindelofSpace
 
 /-- A continuous image of a Lindelöf set is a Lindelöf set within the codomain. -/
 theorem LindelofSpace.of_continuous_surjective {f : X → Y} [LindelofSpace X] (hf : Continuous f)
     (hsur : Function.Surjective f) : LindelofSpace Y where
   isLindelof_univ := by
     rw [← Set.image_univ_of_surjective hsur]
-    exact IsLindelof.image (isLindelof_univ_iff.mpr ‹_›) hf
+    exact IsLindelof.image isLindelof_univ hf
 
-/-- A set `s` is Hereditarily Lindelöf if every subset is a Lindelof set. We require this only
-for open sets in the definition, and then conclude that this holds for all sets by ADD. -/
+/-- A set `s` is Hereditarily Lindelöf if every subset is a Lindelof set. -/
 def IsHereditarilyLindelof (s : Set X) :=
   ∀ t ⊆ s, IsLindelof t
 
+lemma IsHereditarilyLindelof.isLindelof_subset (hs : IsHereditarilyLindelof s) (ht : t ⊆ s) :
+    IsLindelof t := hs t ht
+
+lemma IsHereditarilyLindelof.isHereditarilyLindelof_subset (hs : IsHereditarilyLindelof s)
+    (ht : t ⊆ s) : IsHereditarilyLindelof t := fun u hu => hs u (hu.trans ht)
+
+lemma IsHereditarilyLindelof.isLindelof (hs : IsHereditarilyLindelof s) :
+    IsLindelof s := hs.isLindelof_subset Subset.rfl
+
+/-- A continuous image of a Hereditarily Lindelöf set is a Hereditarily Lindelöf set. -/
+theorem IsHereditarilyLindelof.image_of_continuousOn {f : X → Y} (hs : IsHereditarilyLindelof s)
+    (hf : ContinuousOn f s) : IsHereditarilyLindelof (f '' s) := by
+  intro t hst
```

Comment:
I would use [Set.subset_image_iff](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Set/Image.html#Set.subset_image_iff) here but it's a matter of taste.